app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ShoppingListController < ApplicationController
+  def index
+    @recipes = Recipe.all
+    @ingredients = RecipeFood.where(recipe_id: @recipes.map(&:id))
+    @foods = Food.where(id: @ingredients.map(&:food_id))
+  end
+
+  def total_price
+    price = 0
+    @ingredients.each do |recipe_food|
+      price += Food.find(recipe_food.food_id).price * recipe_food.quantity
+    end
+    price
+  end
+
+  helper_method :total_price
```

Comment:
- You could've used the helpers' files created by rails for adding your helper methods. 🙏 This will help to maintain your code.